docs: move rkt to external drivers (#7662)
hashicorp/nomad: website/pages/docs/drivers/external/rkt.mdx

other projects. Project activity has declined and there are unpatched CVEs.
The project has been [archived by the CNCF](https://github.com/rkt/rkt/issues/4004#issuecomment-507358362)

Nomad 0.11 converted the rkt driver to an external driver. We will not prioritize features

> or pull requests that affect the rkt driver. The external driver will be available as an open source
> repository for community ownership.

Reviewer: ```suggestion
or pull requests that affect the rkt driver. The external driver is available as an [open source
repository](https://github.com/hashicorp/nomad-driver-rkt) for community ownership.
```
